controllers/vsl.go

```diff
@@ -122,41 +122,38 @@ func (r *DPAReconciler) ValidateVolumeSnapshotLocations(dpa oadpv1alpha1.DataPro
 			Spec: *vslSpec.Velero,
 		}
 
+		// check if VSL name is specified in DPA spec
+		if vslSpec.Name != "" {
+			vsl.Name = vslSpec.Name
+		}
+
 		// check for valid provider
 		if vslSpec.Velero.Provider != AWSProvider && vslSpec.Velero.Provider != GCPProvider &&
 			vslSpec.Velero.Provider != Azure {
 			r.Log.Info("Non-supported provider specified, might be a misconfiguration")
 
-			r.EventRecorder.Event(&vsl,
-				corev1.EventTypeWarning,
-				"VSL provider is invalid",
-				fmt.Sprintf("VSL provider %s is invalid, might be a misconfiguration", vslSpec.Velero.Provider),
-			)
+			return false, fmt.Errorf("invalid provider %s for VSL %s", vslSpec.Velero.Provider, vsl.Name)
 		}
 
 		//AWS
 		if vslSpec.Velero.Provider == AWSProvider {
 			//in AWS, region is a required field
 			if len(vslSpec.Velero.Config[AWSRegion]) == 0 {
-				return false, errors.New("region for AWS VSL is not configured, please ensure a region is configured")
+				return false, fmt.Errorf("region for AWS VSL %s is not configured, please ensure a region is configured", vsl.Name)
 			}
 
 			// check for invalid config key
 			for key := range vslSpec.Velero.Config {
 				valid := validAWSKeys[key]
 				if !valid {
-					return false, fmt.Errorf("%s is not a valid AWS config value", key)
+					return false, fmt.Errorf("%s is not a valid AWS config value for VSL %s", key, vsl.Name)
 				}
 			}
 			//checking the aws plugin, if not present, throw warning message
 			if !containsPlugin(dpa.Spec.Configuration.Velero.DefaultPlugins, AWSProvider) {
 				r.Log.Info("VSL for AWS specified, but AWS plugin not present, might be a misconfiguration")
 
-				r.EventRecorder.Event(&vsl,
-					corev1.EventTypeWarning,
-					"VolumeSnapshotLocation is invalid",
-					fmt.Sprintf("could not validate vsl for AWS plugin on: %s/%s", vsl.Namespace, vsl.Name),
-				)
+				return false, fmt.Errorf("VSL %s for AWS specified, but AWS plugin not present, might be a misconfiguration", vsl.Name)
```

Comment:
we do not have this kind of validation for BSL, right? Should we add it (if yes, another PR)

Reply:
hmm, probably not, will confirm and create a PR if not. Noted.